File: Services/ApplicationAuthorization/TixFactory.ApplicationAuthorization/Operations/GetApplicationOperation.cs
using System;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using TixFactory.ApplicationAuthorization.Entities;
using TixFactory.Operations;

namespace TixFactory.ApplicationAuthorization
{
	internal class GetApplicationOperation : IAsyncOperation<string, ApplicationResult>
	{
		private readonly IApplicationEntityFactory _ApplicationEntityFactory;
		private readonly IOperationEntityFactory _OperationEntityFactory;

		public GetApplicationOperation(IApplicationEntityFactory applicationEntityFactory, IOperationEntityFactory operationEntityFactory)
		{
			_ApplicationEntityFactory = applicationEntityFactory ?? throw new ArgumentNullException(nameof(applicationEntityFactory));
			_OperationEntityFactory = operationEntityFactory ?? throw new ArgumentNullException(nameof(operationEntityFactory));
		}

		public async Task<(ApplicationResult output, OperationError error)> Execute(string applicationName, CancellationToken cancellationToken)
		{
			var application = _ApplicationEntityFactory.GetApplicationByName(applicationName);

			ApplicationResult result = null;
			if (application != null)
			{
				var operations = _OperationEntityFactory.GetOperations(application.Id);

				result = new ApplicationResult
				{
					Name = application.Name,
					Operations = operations.Select(o => new OperationResult
					{
						Name = o.Name,
						Enabled = o.Enabled,
						Created = o.Created,
						Updated = o.Updated
					}).ToArray(),
					Created = application.Created,
					Updated = application.Updated
				};
			}

			return (result, null);
		}
	}
}

File: Node/application-authorization-service/operations/GetApplicationOperation.js
export default class {
	constructor(applicationEntityFactory, operationEntityFactory) {
		this.applicationEntityFactory = applicationEntityFactory;
		this.operationEntityFactory = operationEntityFactory;
	}

    get name() {
        return "GetApplication";
    }
	
	get requestParameters() {
		return ["data"];
	}
 
    async execute(requestBody) {
		const application = await this.applicationEntityFactory.getOrCreateApplicationByName(requestBody.data);
		if (!application) {
			return Promise.resolve(null);
		}

		const operations = await this.operationEntityFactory.getOperationsByApplicationName(application.name);
		return Promise.resolve({
			name: application.name,
			operations: operations.map(o => {
				return {
					name: o.name,
					enabled: o.enabled,
					created: o.created,
					updated: o.updated
				}
			}),
			created: application.created,
			updated: application.updated
		});
    }
};